utils.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+func userAndRepoFromURL(repoURL string) (string, string, error) {
+	u, err := url.ParseRequestURI(repoURL)
+	if err != nil {
+		return "", "", err
+	}
+
+	pathSplit := strings.Split(u.Path, "/")
```

Comment:
I think it is better to use [`path.Split`](https://godocs.io/path#Split) here as it will work on non-Unix systems as well.